src/modules/0.0.3/aws_acm_import/index.ts

```diff
@@ -0,0 +1,42 @@
+import { ImportCertificateCommandInput } from '@aws-sdk/client-acm';
+import { AWS, } from '../../../services/gateways/aws';
+import { Context, Crud, Mapper, Module, } from '../../interfaces'
+import { AwsAcmListModule } from '../aws_acm_list';
+import { CertificateImport } from './entity';
+import * as metadata from './module.json';
+
+
+export const AwsAcmImportModule: Module = new Module({
+  ...metadata,
+  mappers: {
+    certificateImport: new Mapper<CertificateImport>({
+      entity: CertificateImport,
+      entityId: (e: CertificateImport) => e.id?.toString() ?? '',
+      equals: () => true, // only database values
+      source: 'db',
+      cloud: new Crud({
+        create: async (es: CertificateImport[], ctx: Context) => {
+          const client = await ctx.getAwsClient() as AWS;
+          const textEncoder = new TextEncoder();
+          for (const e of es) {
+            const input: ImportCertificateCommandInput = {
+              Certificate: textEncoder.encode(e.body),
+              PrivateKey: textEncoder.encode(e.privateKey),
+            };
+            if (e.chain) {
+              input.CertificateChain = textEncoder.encode(e.chain);
+            }
+            const importedCertArn = await client.importCertificate(input);
+            if (!importedCertArn) throw new Error('Error importing certificate');
```

Comment:
Perhaps import failure should flag the record with a text column on the row? (And if so, then any record with that column not null should be ignored by this logic.)

Reply:
Hmm I see your point. Could be interesting. Do you think it could mislead users thinking that since it does not errored everything worked fine?

Reply:
Well, blocking the `apply` with an error is probably better. I guess it'd be good if the error message had more information about what went wrong?

Reply:
well that would be happen if `apply` fails. This specific line is just one of those `What!?` or `this should never happen` errors... I'm just making the check for sanity and to let typescript know that `importedCertArn` will be defined, but if import fails it will simply `throw`.

Reply:
> well that would be happen if `apply` fails. This specific line is just one of those `What!?` or `this should never happen` errors... I'm just making the check for sanity and to let typescript know that `importedCertArn` will be defined, but if import fails it will simply `throw`.

Well, so, if the cert definition provided by the user is bad, I would expect `importCertificate` to fail? And hopefully with a useful error message from AWS that the certificate is invalid in some way. I'd rather it be wrapped in a `try-catch` and then re-throw the error with any extra explanatory text we want to prepend to their message.

Reply:
Yes `importCertificate` will fail and `throw`. The current errors from AWS are ok, let me share them and see if we think is necessary to add more detail, but I don't think so.

![image](https://user-images.githubusercontent.com/5898639/167697759-de46794f-2cb5-4060-af3c-7322464e6fd6.png)

![image](https://user-images.githubusercontent.com/5898639/167697942-92ee2695-aaef-4eb2-a595-b75267f104b0.png)

![image](https://user-images.githubusercontent.com/5898639/167698291-d6a94ae4-29aa-4922-849d-67c231333f38.png)

Reply:
That's pretty good.

There's a more generic change I want to make to annotate the `APPLY` errors with the table and ID value that triggered said error, but that's outside of the scope of this PR.

Approved!